src/kqueue.rs

```diff
@@ -86,11 +86,11 @@ impl Poller {
     /// # Safety
     ///
     /// The file descriptor must be valid and it must last until it is deleted.
-    pub unsafe fn add(&self, fd: RawFd, ev: Event, mode: PollMode) -> io::Result<()> {
+    pub fn add(&self, fd: RawFd, ev: Event, mode: PollMode) -> io::Result<()> {
         self.add_source(SourceId::Fd(fd))?;
 
         // File descriptors don't need to be added explicitly, so just modify the interest.
-        self.modify(BorrowedFd::borrow_raw(fd), ev, mode)
+        self.modify(unsafe { BorrowedFd::borrow_raw(fd) }, ev, mode)
```

Comment:
Since the responsibility for ensuring safety has shifted from the caller to us, I feel we need to leave a comment here explaining that this is safe. (Something like `SAFETY: See https://github.com/smol-rs/polling/issues/255.` is probably fine, though.)